feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewMode.kt

```diff
@@ -44,4 +44,13 @@ sealed interface PreviewMode {
         val videoCaptureUri: Uri?,
         val onVideoCapture: (PreviewViewModel.VideoCaptureEvent) -> Unit
     ) : PreviewMode
+
+    /**
+     * Under this mode, the app is launched by an external intent to capture multiple images.
+     */
+    data class ExternalMultipleImageCaptureMode(
+        val imageCaptureUris: List<Uri>?,
+        var currentUriIndex: Int,
```

Comment:
Let's avoid the "Mode" containing mutable state. Can you just keep track of the `currentUriIndex` in an instance variable inside of `PreviewViewModel`?

Reply:
It'll have to go into PreviewUiState so that UI component get use it to get the right uri